api/v1/api_v1_user_groups.go

```diff
@@ -0,0 +1,299 @@
+/*
+Copyright (c) 2023 Dell Inc, or its subsidiaries.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package v1
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/dell/goisilon/api"
+)
+
+// GetIsiGroup queries the group by group-id.
+func GetIsiGroup(ctx context.Context, client api.Client, groupName *string, gid *int32) (group *IsiGroup, err error) {
+
+	// PAPI call: GET https://1.2.3.4:8080/platform/1/auth/groups/<group-id>
+
+	authGroupId, err := getAuthMemberId(fileGroupTypeGroup, groupName, gid)
+	if err != nil {
+		return
+	}
+
+	var groupListResp *isiGroupListResp
+	if err = client.Get(ctx, groupPath, authGroupId, nil, nil, &groupListResp); err != nil {
+		return
+	}
+
+	if groupListResp.Groups != nil && len(groupListResp.Groups) > 0 {
+		group = groupListResp.Groups[0]
+		return
+	}
+
+	return nil, fmt.Errorf("group not found: %s", authGroupId)
+}
+
+// GetIsiGroupList queries all groups on the cluster with filter,
+// filter by namePrefix, domain, zone, provider, cached, resolveNames, memberOf, zone and limit.
+func GetIsiGroupList(ctx context.Context, client api.Client,
```

Comment:
can we convert the parameters of this function into a struct

Reply:
@chimanjain, in this way, user just need fill nil in the function; else if covert to struct, user need import this struct and fill it.
And also to keep consistent, other functions are using this way, not using struct.

Reply:
ok

Reply:
thanks @chimanjain